dev/build_release.py

```diff
@@ -339,42 +313,118 @@ def start_container_build(self, name):
-          'cd "{gradle_root}"'
-          '; gcloud container builds submit --account={account} --project={project} --config="../{name}-gcb.yml" .'
-        .format(gradle_root=gradle_root, name=name, account=self.__gcb_service_account,
-                project=self.__options.gcb_project),
-        logfile='{name}-gcb-build.log'.format(name=name)
-      )
+      print 'Build/publishing container image for {name} with GCB...'.format(name=name)
+      self.__gcb_build(name, gradle_root, self.__options.gcb_service_account, self.__options.gcb_project)
+    elif self.__options.container_builder == 'gcb-trigger':
+      print 'Build/publishing container image for {name} with a triggered GCB build...'.format(name=name)
+      self.__gcb_trigger_build(name,
+                               gradle_root,
+                               self.__options.gcb_service_account,
+                               self.__options.gcb_service_account_json,
+                               self.__options.gcb_project,
+                               self.__options.gcb_mirror_base_url)
     elif self.__options.container_builder == 'docker':
-      return BackgroundProcess.spawn(
-          'Build/publishing container image for {name} with Docker...'.format(
-              name=name),
-          'cd "{gradle_root}"'
-          ' ; docker build -f Dockerfile -t $(cat ../{name}-docker.yml) .'
-          ' ; docker push $(cat ../{name}-docker.yml)'
-          .format(gradle_root=gradle_root, name=name)
-      )
+      print 'Build/publishing container image for {name} with Docker...'.format(name=name)
+      self.__docker_build(name, gradle_root)
     else:
       raise NotImplemented(
           'container_builder="{0}"'.format(self.__options.container_builder))
 
+  @classmethod
+  def __gcb_build(cls, name, gradle_root, gcb_service_account, gcb_project):
+    # Local .gradle dir stomps on GCB's .gradle directory when the gradle
+    # wrapper is installed, so we need to delete the local one.
+    # The .gradle dir is transient and will be recreated on the next gradle
+    # build, so this is OK.
+    gradle_cache = '{name}/.gradle'.format(name=name)
+    if os.path.isdir(gradle_cache):
+      # Tell rmtree to delete the directory even if it's non-empty.
+      shutil.rmtree(gradle_cache)
+    print 'Building and publishing container image via GCB for {name}...'.format(name=name)
+    cmds = [
+      ('gcloud container builds submit --account={account} --project={project} --config"../{name}-gcb.yml"'
+       .format(name=name, account=gcb_service_account, project=gcb_project))
+    ]
+    logfile = '{name}-gcb-build.log'.format(name=name)
+    run_shell_and_log(cmds, logfile, cwd=gradle_root)
+
+  @classmethod
+  def __gcb_trigger_build(cls, name, gradle_root, gcb_service_account, gcb_service_account_json, gcb_project, mirror_base_url):
+    tag = run_quick('cat {name}-gcb-trigger.yml'.format(name=name), echo=False).stdout.strip()
```

Comment:
how big is tag? run_quick is subject to deadlock if it produces more output than the buffer.